src/precision.cpp

```diff
@@ -0,0 +1,225 @@
+        {Precision::FP64, "FP64"},
+        {Precision::FP32, "FP32"},
+        {Precision::FP16, "FP16"},
+        {Precision::I64, "I64"},
+        {Precision::I32, "I32"},
+        {Precision::I16, "I16"},
+        {Precision::I8, "I8"},
+        {Precision::I4, "I4"},
+        {Precision::U64, "U64"},
+        {Precision::U32, "U32"},
+        {Precision::U16, "U16"},
+        {Precision::U8, "U8"},
+        {Precision::U4, "U4"},
+        {Precision::U1, "U1"},
+        {Precision::MIXED, "MIXED"},
+        {Precision::Q78, "Q78"},
+        {Precision::BIN, "BIN"},
+        {Precision::BOOL, "BOOL"},
+        {Precision::UNDEFINED, "UNDEFINED"},
+        {Precision::CUSTOM, "CUSTOM"}};
+    auto it = precisionMap.find(precision);
+    if (it == precisionMap.end()) {
+        return "DT_INVALID";  // TODO other way? why translate it to TF equivalent maybe UNDEFINED?
+    }
+    return it->second;
+}
+
+Precision fromString(const std::string& s) {
+    static std::unordered_map<std::string, Precision> precisionMap{
+        {"BF16", Precision::BF16},
+        {"FP64", Precision::FP64},
+        {"FP32", Precision::FP32},
+        {"FP16", Precision::FP16},
+        {"I64", Precision::I64},
+        {"I32", Precision::I32},
+        {"I16", Precision::I16},
+        {"I8", Precision::I8},
+        {"I4", Precision::I4},
+        {"U64", Precision::U64},
+        {"U32", Precision::U32},
+        {"U16", Precision::U16},
+        {"U8", Precision::U8},
+        {"U4", Precision::U4},
+        {"U1", Precision::U1},
+        {"MIXED", Precision::MIXED},
+        {"Q78", Precision::Q78},
+        {"BIN", Precision::BIN},
+        {"BOOL", Precision::BOOL},
+        {"UNDEFINED", Precision::UNDEFINED},
+        {"CUSTOM", Precision::CUSTOM}};
+    auto it = precisionMap.find(s);
+    if (it == precisionMap.end()) {
+        return Precision::UNDEFINED;
+    }
+    return it->second;
+}
+
+Precision kfsPrecisionToOvmsPrecision(const std::string& s) {
+    static std::unordered_map<std::string, Precision> precisionMap{
+        {"BOOL", Precision::BOOL},
+        {"FP64", Precision::FP64},
+        {"FP32", Precision::FP32},
+        {"FP16", Precision::FP16},
+        {"INT64", Precision::I64},
+        {"INT32", Precision::I32},
+        {"INT16", Precision::I16},
+        {"INT8", Precision::I8},
+        {"UINT64", Precision::U64},
+        {"UINT32", Precision::U32},
+        {"UINT16", Precision::U16},
+        // {"BYTES", Precision::??}, // TODO decide what to do with BYTES
+        {"UINT8", Precision::U8}};
+    auto it = precisionMap.find(s);
+    if (it == precisionMap.end()) {
+        return Precision::UNDEFINED;
+    }
+    return it->second;
+}
+
+size_t KFSDataTypeSize(const KFSDataType& datatype) {
+    static std::unordered_map<KFSDataType, size_t> datatypeSizeMap{
+        {"BOOL", 1},
+        {"UINT8", 1},
+        {"UINT16", 2},
+        {"UINT32", 4},
+        {"UINT64", 8},
+        {"INT8", 1},
+        {"INT16", 2},
```

Comment:
KFS might have similar method already implemented

Reply:
I did not see anything like that.